Sources/NIOCore/EventLoopFuture+AssumeIsolated.swift

```diff
@@ -14,8 +14,14 @@
 
 /// A struct wrapping an ``EventLoop`` that ensures all calls to any method on this struct
 /// are coming from the event loop.
-@usableFromInline
-struct IsolatedEventLoop {
+///
+/// This type is explicitly not `Sendable`. It may only be constructed on an event loop,
+/// using ``EventLoop/assumeIsolated()``, and may not subsequently be passed to other isolation
+/// domains.
```

Comment:
I think other isolation domains is actually too weak of a statement. This type must not be held inside any asynchronous context.

Reply:
I think that's not appropriate either: non-asynchronous contexts are themselves dangerous. As task executors aren't supported today, I think the statement as written is correct. I would now reject any plan to _add_ task executors until these issues are resolved, so I'm inclined to keep the text.

Reply:
What non-asynchronous context are you referring to? I consider inside an actor an asynchronous context since you need to be inside a `Task`.